tests/test_mean_estimator_boosting.py

```diff
@@ -0,0 +1,200 @@
+def weighted_source(weighted_volume):
+    src = Simulation(
+        n=400,
+        vols=weighted_volume,
+        offsets=0,
+        amplitudes=1,
+        seed=SEED,
+        dtype=weighted_volume.dtype,
+    )
+
+    return src
+
+
+# MeanEstimator Tests.
+def test_fsc(source, estimated_volume):
+    """Compare estimated volume to source volume with FSC."""
+    # Fourier Shell Correlation
+    fsc_resolution, fsc = source.vols.fsc(estimated_volume, pixel_size=1, cutoff=0.5)
+
+    # Check that resolution is less than 2.1 pixels.
+    np.testing.assert_array_less(fsc_resolution, 2.1)
+
+    # Check that second to last correlation value is high (>.90).
+    np.testing.assert_array_less(0.90, fsc[0, -2])
+
+
+def test_mse(source, estimated_volume):
+    """Check the mean-squared error between source and estimated volumes."""
+    mse = np.mean((source.vols.asnumpy() - estimated_volume.asnumpy()) ** 2)
+    np.testing.assert_allclose(mse, 0, atol=1e-3)
+
+
+def test_total_energy(source, estimated_volume):
+    """Test that energy is preserved in reconstructed volume."""
+    og_total_energy = np.sum(source.vols)
+    recon_total_energy = np.sum(estimated_volume)
+    np.testing.assert_allclose(og_total_energy, recon_total_energy, rtol=1e-3)
+
+
+def test_boost_flag(source, estimated_volume):
+    """Manually boost a source and reconstruct without boosting."""
+    ims = source.projections[:]
+    rots = source.rotations
+    sym_rots = source.symmetry_group.matrices
+    sym_order = len(sym_rots)
+
+    # Manually boosted images and rotations.
+    ims_boosted = np.tile(ims, (sym_order, 1, 1))
+    rots_boosted = np.zeros((sym_order * source.n, 3, 3), dtype=source.dtype)
+    for i, sym_rot in enumerate(sym_rots):
+        rots_boosted[i * source.n : (i + 1) * source.n] = sym_rot @ rots
+    rots_boosted = Rotation(rots_boosted)
+
+    # Manually boosted source.
+    boosted_source = ArrayImageSource(ims_boosted, angles=rots_boosted.angles)
+
+    # Estimate volume with boosting OFF.
+    basis = FFBBasis3D(boosted_source.L, dtype=boosted_source.dtype)
+    estimator = MeanEstimator(boosted_source, basis, boost=False)
+    est_vol = estimator.estimate()
+
+    # Check reconstructions are equal.
+    mse = np.mean((estimated_volume.asnumpy() - est_vol.asnumpy()) ** 2)
+    np.testing.assert_allclose(mse, 0, atol=utest_tolerance(source.dtype))
+
+
+# WeightVolumesEstimator Tests.
+def test_weighted_volumes(weighted_source):
+    """
+    Test WeightedVolumeEstimator reconstructs multiple volumes using symmetry boosting.
+    """
+    src = weighted_source
+
+    # Use source states to assign weights to volumes.
+    weights = np.zeros((src.n, src.C), dtype=src.dtype)
+    weights[:, 0] = abs(src.states - 1.99)  # sends states [1, 2] to weights [.99, .01]
+    weights[:, 1] = abs(-src.states + 1.01)  # sends states [1, 2] to weights [.01, .99]
+
+    # Scale weights
+    n0 = (-src.states + 2).sum()  # number of images from vol[0]
```

Comment:
This is more clever than it needs to be.  Can use sum() of Trues or make really obvious with count.

n0  = np.count_nonzero(src.states == 1)
n1 = np.count_nonzero(src.states == 2)

...or you can use a loop over src.C... or vector ...     Probably count, is a good compromise, reads real nice.

Reply:
Good call, count is much clearer! Thanks. Changed.